bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/part/MultiPageEditorPart.java

```diff
@@ -149,10 +154,31 @@ public abstract class MultiPageEditorPart extends EditorPart implements IPageCha
 	private ListenerList<IPageChangedListener> pageChangeListeners = new ListenerList<>(ListenerList.IDENTITY);
 
 	/**
-	 * Creates an empty multi-page editor with no pages.
+	 * Creates an empty multi-page editor with no pages and registers a
+	 * {@link PropertyChangeListener} to listen for changes to the editor's
+	 * preference..
 	 */
 	protected MultiPageEditorPart() {
 		super();
+		getAPIPreferenceStore().addPropertyChangeListener(event -> {
+			handlePropertyChange(event);
+		});
+	}
+
+	/**
+	 * Handles property change events related to editor preferences.
+	 *
+	 * <p>
+	 * This method is invoked when a property change occurs in the preference store.
+	 * </p>
+	 *
+	 * @param event the {@link PropertyChangeEvent} triggered by a change in the
+	 *              preference store
+	 */
+	private void handlePropertyChange(PropertyChangeEvent event) {
```

Comment:
```suggestion
	protected boolean handlePropertyChange(PropertyChangeEvent event) {
```

I would let this method simpyl return if an update is required and handle the update container in the event listener.